samples/shields/x_nucleo_nfc04a1/src/main.c

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2023 Kickmaker
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/drivers/eeprom.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/eeprom/eeprom_st25dv.h>
+#include <stdio.h>
+
+/* 1000 msec = 1 sec */
+#define SLEEP_TIME_MS   1000
+
+#define EEPROM_SAMPLE_OFFSET 0
+#define EEPROM_SAMPLE_MAGIC  0xEE9704
+
+#define LED1_NODE DT_ALIAS(shield_green_led1)
+#define LED2_NODE DT_ALIAS(shield_blue_led2)
+#define LED3_NODE DT_ALIAS(shield_yellow_led3)
+
+struct perisistant_values {
+	uint32_t magic;
+	uint32_t boot_count;
+};
+
+static const struct gpio_dt_spec led1 = GPIO_DT_SPEC_GET(LED1_NODE, gpios);
+static const struct gpio_dt_spec led2 = GPIO_DT_SPEC_GET(LED2_NODE, gpios);
+static const struct gpio_dt_spec led3 = GPIO_DT_SPEC_GET(LED3_NODE, gpios);
+
+
+/*
+ * Get a device structure from a devicetree node with alias eeprom-0
+ */
+static const struct device *get_eeprom_device(void)
+{
+	const struct device *const dev = DEVICE_DT_GET(DT_ALIAS(eeprom_0));
+
+	if (!device_is_ready(dev)) {
+		printk("\nError: Device \"%s\" is not ready; "
+		       "check the driver initialization logs for errors.\n",
+		       dev->name);
+		return NULL;
+	}
+
+	printk("Found EEPROM device \"%s\"\n", dev->name);
+	return dev;
+}
+
+int main(void)
+{
+	const struct device *eeprom = get_eeprom_device();
+	size_t eeprom_size;
+	struct perisistant_values values;
+	int rc;
+	int ret;
+
+	if (!gpio_is_ready_dt(&led1) || !gpio_is_ready_dt(&led2) || !gpio_is_ready_dt(&led3)) {
+		return 0;
+	}
+
+	ret = gpio_pin_configure_dt(&led1, GPIO_OUTPUT_ACTIVE);
+	if (ret < 0) {
+		return 0;
+	}
+	ret = gpio_pin_configure_dt(&led2, GPIO_OUTPUT_ACTIVE);
+	if (ret < 0) {
+		return 0;
+	}
+	ret = gpio_pin_configure_dt(&led3, GPIO_OUTPUT_ACTIVE);
+	if (ret < 0) {
+		return 0;
+	}
+
+	if (eeprom == NULL) {
+		return 0;
+	}
```

Comment:
No error output on the console would make initialization errors unnecessaril hard to debug.